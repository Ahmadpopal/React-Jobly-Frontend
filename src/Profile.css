.Profile-title {
  color: black;
  text-align: center;
}

.Profile-user-info {
  margin-top: 10px;
  padding: 5px;
  border: 1px solid black;
  background-color: rgb(206, 206, 206);
  color: white;
}
