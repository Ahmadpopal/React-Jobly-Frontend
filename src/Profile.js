import React, { useContext} from 'react';
import UserContext from './Context'
import { Container, Row, Col } from 'reactstrap';


import './Profile.css'



const Profile = () => {

    const { currentUser } = useContext(UserContext);
    
    return (
        

     <>
        <Container>
        <Row>
                <Col md={{ size: 6, offset: 3 }} className="Profile-user-info" xs="6">
                <h3 className="Profile-title">Profile</h3>
                </Col>
            </Row>
            <Row>
                <Col md={{ size: 6, offset: 3 }} className="Profile-user-info" xs="6">
                    <h4>UserName: {currentUser.username}</h4>
                </Col>
            </Row>
            <Row>
                <Col md={{ size: 6, offset: 3 }} className="Profile-user-info" xs="6">
                    <h4>First Name: {currentUser.firstName}</h4>
                </Col>
            </Row>
            <Row>
                <Col md={{ size: 6, offset: 3 }} className="Profile-user-info" xs="6">
                    <h4>Last Name: {currentUser.lastName}</h4>
                </Col>
            </Row>
            <Row>
                <Col md={{ size: 6, offset: 3 }} className="Profile-user-info" xs="6">
                    <h4>Email: {currentUser.email}</h4>
                </Col>
            </Row>
        </Container>

    </>
    )
}





export default Profile