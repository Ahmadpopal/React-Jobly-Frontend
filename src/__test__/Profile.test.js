import React from 'react';
import UserContext from '../Context';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Profile from '../Profile';


describe('PROFILE TESTS', () => {
    // SMOKE TEST 
    it('Profile Component should render without crush', () => {
        let currentUser = {
			username   : 'test',
			first_name : 'test',
			last_name  : 'test',
			email      : 'test',
			logo_url   : 'test'
		};

        render(
			<UserContext.Provider value={{ currentUser }}>
				<MemoryRouter>
					<Profile />
				</MemoryRouter>
			</UserContext.Provider>
		);
	});

    // SNAPSHOT TEST 
    it('matches snapshot', () => {
		let currentUser = {
			username   : 'test',
			first_name : 'test',
			last_name  : 'test',
			email      : 'test',
			logo_url   : 'test'
		};

		const { asFragment } = render(
			<UserContext.Provider value={{ currentUser }}>
				<MemoryRouter>
					<Profile />
				</MemoryRouter>
			</UserContext.Provider>
		);
		expect(asFragment()).toMatchSnapshot();
	});



    it('shows user Profile', () => {
		let currentUser = {
			username   : 'test',
			first_name : 'test',
			last_name  : 'test',
			email      : 'test'
		};

		const { getByText } = render(
			<UserContext.Provider value={{ currentUser }}>
				<MemoryRouter>
					<Profile />
				</MemoryRouter>
			</UserContext.Provider>
		);
		expect(getByText('Profile')).toBeInTheDocument();
	});
})