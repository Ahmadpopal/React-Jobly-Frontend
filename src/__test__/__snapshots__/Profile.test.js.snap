// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PROFILE TESTS matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="Profile-user-info col-6 col-md-6 offset-md-3"
      >
        <h3
          class="Profile-title"
        >
          Profile
        </h3>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="Profile-user-info col-6 col-md-6 offset-md-3"
      >
        <h4>
          UserName: test
        </h4>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="Profile-user-info col-6 col-md-6 offset-md-3"
      >
        <h4>
          First Name: 
        </h4>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="Profile-user-info col-6 col-md-6 offset-md-3"
      >
        <h4>
          Last Name: 
        </h4>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="Profile-user-info col-6 col-md-6 offset-md-3"
      >
        <h4>
          Email: test
        </h4>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
